scripts/common/auth.py

```
@@ -19,6 +20,35 @@
+
+    Some P21 middleware instances return XML instead of JSON for token
+    endpoints. This tries JSON first, then falls back to XML regex parsing.
+    """
+    text = response.text
+    # Try JSON first
+    try:
+        data = response.json()
+        if isinstance(data, dict) and ("AccessToken" in data or "access_token" in data):
+            return data
+    except Exception:
+        pass
```

Ruff (0.15.0)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)